### PR TITLE
enhance(kevel): extract libs/pong + support more formats

### PR DESCRIPTION
**Pull request by fiji-flo**
## Summary
Support more formats for in the kevel lambda and factor out most of the logic into libs.

### Problem

The lambda only supported one format (type and not a list).

### Solution

Support requesting and returning multiple formats.

---

## Screenshots

`n/a`

## How did you test this change?

Ran tests in the stage lambda.

### CODE REVIEW

File: deployer/aws-lambda/kevel/index.js [hunk 1]
```diff
@@ -1,51 +1,34 @@
...
-  if (hmac.digest("base64") == digest) {
-    // === won't work...
-    return s;
-  }
-  return null;
-}
+const CODER = new Coder(SIGN_SECRET);
+const pongGetHandler = makePongGetHandler(client, CODER, env);
+const pongClickHandler = makePongClickHandler(CODER);
+const pongViewedHandler = makePongViewedHandler(CODER);
```
Comment (caugner) [inline deployer/aws-lambda/kevel/index.js:31]:
```suggestion
const handleGet = createPongGetHandler(client, CODER, env);
const handleClick = createPongClickHandler(CODER);
const handleViewed = createPongViewedHandler(CODER);
```

File: libs/pong/click.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/* global fetch */
+export function makePongClickHandler(CODER) {
```
Comment (caugner) [inline libs/pong/click.js:2]:
```suggestion
export function makePongClickHandler(coder) {
```

File: libs/pong/coding.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { createHmac } from "node:crypto";
+
+export class Coder {
+  signSecret;
+  constructor(signSecret) {
```
Comment (caugner) [inline libs/pong/coding.js:5]: Can we add JSDoc to declare the field + param type?

File: libs/pong/index.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export * from "./coding.js";
```
Comment (caugner) [inline libs/pong/index.js:1]: For the GCP function we would need a good `index.d.ts` file, but it might be easier to create it then.

**Review by caugner — CHANGES_REQUESTED** [review]
LGTM (did not test), I would just suggest to rename some methods/variables and to move `cc2ip` into `libs/pong`.

File: libs/pong/get.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+const PLACEMENTS = {
+  banner: 369,
+  topBanner: 585,
+};
+
+export function createPongGetHandler(client, coder, env) {
+  const {
+    CARBON_ZONE_KEY,
+    FALLBACK_ENABLED,
+    // eslint-disable-next-line n/no-missing-import
```
Comment (caugner) [inline libs/pong/get.js:15]:
```suggestion
```

File: libs/pong/package.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "@yari-internal/pong",
+  "version": "0.0.1",
+  "private": true,
+  "license": "MPL-2.0",
+  "type": "module",
+  "main": "index.js",
+  "dependencies": {
```
Comment (caugner) [inline libs/pong/package.json:8]: Do we not have a `yarn.lock` file here intentionally?

**Review by caugner — CHANGES_REQUESTED** [review]
Is there any chance you could split this PR into two PRs, where one "supports more formats" and the other one extracts the core logic to `libs/pong`?

File: libs/pong/package.json (hunk 6 above)
Comment (fiji-flo) [reply libs/pong/package.json:8]: nope, just never had to run yarn here

**Review by caugner — APPROVED** [review]

**Pull request merged**